checkers/rules/rules.go

```diff
@@ -771,3 +771,19 @@ func stringsCompare(m dsl.Matcher) {
 		`strings.Compare($s1, $s2) > 0`).
 		Suggest(`$s1 > $s2`)
 }
+
+//doc:summary   Finds unchecked errors in if statements
+//doc:tags      diagnostic experimental
+//doc:before    if err := expr(); err2 != nil { /*...*/ }
+//doc:after     if err := expr(); err != nil { /*...*/ }
+func errCheckInIf(m dsl.Matcher) {
+	m.Match(
+		`if $err := $_($*_); $err2 != nil { $*_ }`,
+		`if $err = $_($*_); $err2 != nil { $*_ }`,
+		`if $*_, $err := $_($*_); $err2 != nil { $*_ }`,
+		`if $*_, $err = $_($*_); $err2 != nil { $*_ }`).
+		Where(m["err"].Type.Implements("error") && m["err2"].Type.Implements("error") &&
+			m["err"].Text != m["err2"].Text).
+		Report("returned error '$err' must be checked").
```

Comment:
How about `"$err error is unchecked, maybe intended to check it instead of $err2"`?
The error message should give a hint about the potentially mistyped err1 vs err2.

Reply:
+